src/main/cpp/states/Music.cpp

```diff
@@ -0,0 +1,107 @@
+#include <states/Music.h>
+
+int Music::GetButton() 
+{
+    for (int i = 1; i < 9; i++) 
+    {
+        if (m_joystick->GetRawButton(i))
+        {
+            return i;
+        }
+    }
+    return 0;
+}
+
+void Music::LoadMusicSelection(int offset)
+{
+    songSelection += offset;
+
+    if (songSelection >= SONG_COUNT)
+    {
+        songSelection = 0;
+    }
+
+    if (songSelection < 0)
+    {
+        songSelection = SONG_COUNT - 1;
+    }
+
+    m_orchestra -> LoadMusic(songs[songSelection]);
+    timeToPlayLoops = 10;
+}
+
+void Music::Init()
+{
+    m_orchestra = new Orchestra();
+
+    m_talon = new TalonFX * [TALON_COUNT];
+
+    for (int i = 0; i < TALON_COUNT; i++)
+    {
+        m_talon[i] = new TalonFX(i+1);
+        m_orchestra -> AddInstrument(*m_talon[i]);
+    }
+    m_joystick = new frc::Joystick(0);
+
+    LoadMusicSelection(0);
+}
+
+void Music::Run()
+{
+    int btn = GetButton();
+    int currentPOV = m_joystick->GetPOV();
+
+    if (timeToPlayLoops > 0)
+    {
+        --timeToPlayLoops;
+        if (timeToPlayLoops == 0)
+        {
+            m_orchestra -> Play();
+        }
+    }
+
+    if (lastButton != btn)
+    {
+        lastButton = btn;
+

```

Comment:
Don't reference joystick.  If you want to provide next / prev song, just provide a method that does that.

If you don't tie this level to a joystick, we can use it anywhere.   We can have a class that uses joystick buttons to call play, shuffle up, shuffle down, etc. to get the functionality you have here.   But we can also call it when we start/finish climbing or shoot balls in auton, etc.